### PR TITLE
add `createMutationContext` to encapsulate state with `mutate` and `afterMutation`

### PR DESCRIPTION
**Pull request by ryanatkn**
Removes `mutate` and `afterMutation` from the `ui` object, clarifying its role as a namespace for data, and hoists them to the `MutationContext`, which now has its state encapsulated in the new `createMutationContext`.

This change means explicitly threading `afterMutation` through helpers that need it. This keeps things simple and explicit in the mutation helpers, if verbose, rather than passing the entire mutation context everywhere, which is too large a surface area to reason about. This also means we'll experience more function signature churn, but the types should keep things tidy.

future

- add `mutate` usage to singular mutations - see `uiMutations.ts`

### CODE REVIEW

File: src/lib/util/mutation.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	return {
+		actionName,
+		params,
+		ui,
+		actions,
+		invoke,
+		mutate,
+		afterMutation,
+	};
+};
```
Comment (ryanatkn) [inline src/lib/util/mutation.ts:74]: moves stuff that was previously in `$lib/ui/mutation.ts` into this `$lib/util` module -- I think this is a good pattern, moving the non-Svelte stuff from `ui/` into `util/`

Comment (greatbacon) [reply src/lib/util/mutation.ts:81]: Yeah I can dig it

File: src/lib/vocab/actor/actorMutationHelpers.ts [hunk 2]
```diff
@@ -43,13 +44,11 @@ export const stashActors = (
 	}
 };
 
-export const evictActors = (ui: WritableUi, actorsToEvict: Set<Writable<ClientActor>>): void => {
```
Comment (greatbacon) [inline src/lib/vocab/actor/actorMutationHelpers.ts:46]: were we not calling evictActors anywhere?

Comment (ryanatkn) [reply src/lib/vocab/actor/actorMutationHelpers.ts:46]: apparently not, only the singular version -- we may want this in the future, but a loop at the callsite might be fine too

**Review by greatbacon — APPROVED** [review]

**Pull request merged**